### DISCUSSION TITLE
Need help with pnp dependency errors

### DISCUSSION THREAD

**Discussion by ziggy6792** [yarnpkg/berry#3716] [Q&A]
I am really struggling with fixing dependency errors, trying to use [packageExtensions](https://yarnpkg.com/configuration/yarnrc#packageExtensions)

I made a [sample repo](https://github.com/ziggy6792/typescript-bunlde-experiments/tree/feature/yarn-berry) (brach `yarn-berry`) and started trying to migrate to pnp 

After running `yarn dlx @yarnpkg/doctor` I get some errors. I have tried to add some packageExtensions in `.yarnrc.yml` but I have not managed to make any of these errors go away.
Actually my project builds, bundles and runs fine but I'm really worried about these errors. I don't understand the risks the cause or how to fix them. And I know I really need to understand this well to be able to use pnp in the future. 
![Screenshot 2021-11-09 at 1 08 10 AM](https://user-images.githubusercontent.com/60313838/140786935-2852b719-7dbc-4d1e-b8b4-b454d23dedf2.png)

Apart from these issues I am facing with pnp everything else is working great, super fast and small package sizes. I really hope I can use this tool, thanks a lot for your hard work.

**Comment by arcanis** [comment]
Note that the doctor is intended to give you *potential issues*. It's always a little difficult to know whether something is truly a problem or not, so it tends to err on the cautious side and overreport rather than underreport. So if things work, that's already quite good!

- The "Strings should avoid" one is one of those overzealous rules. It tells you if you have the `node_modules` string anywhere in your project, as they are sometimes used in dangerous ways (like `path.join(__dirname, 'node_modules/dep/index.js')`) for example. Of course there are reasons why you'd still use them, so ignore as needed.

- "Unmet transitive peer dependency" means that `ts-jest` has a peer-dependency on `jest`, but `jest` isn't in your project. Probably something you may want to fix, along with the others.

**Reply by ziggy6792** [reply]
Hi @arcanis 

Thanks a lot for your reply.
Yes the strings one I am not worried about.
But all the peer dependency ones I should be able to cleanup and make go away right?

Take this error as example
`Unmet transitive peer dependency on jest@^27.0.0, via ts-jest@^27.0.5`

The fix here should be what? Because I tried the following and I still get the error. 
- Step 1 (I assume this is needed)
`package.json > devDependencies`
Added ` "jest": "^27.0.0",` 

- Step 2 (I'm not sure this is needed if step 1 is done)
`.yarnrc.yml > packageExtensions`
Added
```
ts-jest@*:
    peerDependencies:
      jest: '*'      
```
(I also tried specifying the exact versions `^27.0.0`/`^27.0.5` too)

I don't understand how to make this error go away (or any of the other errors I have).

Ideally I would have 0 errors, and I will run this doctor every time I add a new package to make sure I don't create issues as I develop. I want to avoid a situation where I have 100s of potential problems and I have to memorise which ones I should not care about.

Thanks a lot

**Reply by ziggy6792** [reply]
I fixed it.
Basically I need if the package complaining is a dependency then you need to add the peer dependency as a dependency to `package.json` too, thats it. 
I didn't need to use `packageExtensions` at all so I am still not sure what I would use them for, maybe that will come clear later.
